improver/metadata/amend.py

```diff
@@ -102,3 +102,35 @@ def set_history_attribute(cube, value, append=False):
         cube.attributes["history"] += "; {}".format(new_history)
     else:
         cube.attributes["history"] = new_history
+
+
+def update_model_id_attr_attribute(cubes, model_id_attr):
+    """Update the dictionary with the unique values of the model_id_attr
+    attribute from within the input cubes. The model_id_attr attribute is
+    expected on all cubes or no cubes.
+
+    Args:
+        cubes (list or iris.cube.CubeList):
+            List of input cubes that might have a model_id_attr attribute.
+        model_id_attr (str):
+            Name of attribute expected on the input cubes. This attribute is
+            expected on the cubes as a space-separated string.
+
+    Returns:
+        dict:
+            Dictionary containing a model_id_attr key, if available.
+
+    Raises:
+        AttributeError: Expected to find the model_id_attr attribute on all
+            cubes or no cubes.
+    """
+    attr_in_cubes = [model_id_attr in c.attributes for c in cubes]
+    if all(attr_in_cubes) != any(attr_in_cubes):
+        msg = f"Expected to find {model_id_attr} attribute on all cubes or no cubes"
+        raise AttributeError(msg)
```

Comment:
I know I asked for the function to do this, but it doesn't make sense any more! We will only reach this test if the user has explicitly specified the `model_id_attr`, therefore, they expect it to be present on all cubes, so the code should reject the case where the attribute is not present on any cube. Please can you change the behaviour and doc-string to do this? (sorry!!)

Reply:
Yes, I've removed the handling of none of the cubes haven't the `model_id_attr` attribute, so that the expectation is that if the `model_id_attr` attribute is specified then it should be present on all cubes.